Simplex/Solver.php

```diff
@@ -17,16 +17,18 @@ class Solver
 
 	/** @var array */
 	private $steps = array();
-
-
-
-	const MAX_STEPS = 16;
+	private $max_steps = 16;
 
 
 
 	/** @param  Task $task */
-	function __construct(Task $task)
+	/** @param int $maxSteps (optional) */
+	function __construct(Task $task, int $maxSteps = null)
```

Comment:
This library requires PHP from 5.3 which doesn't support scalar typehints, please remove it.